FogCreek/Glitch-Community
# Update collection page to handle extra memory

## Links
* Remix link: https://sarah-staging.glitch.me/
* Issue-tracker link: https://app.clubhouse.io/glitch/story/6410/update-collection-page-to-handle-extra-memory-collections
* Specs/doc links (in clubhouse)

## GIF/Screenshots:
<img width="1172" alt="Screen Shot 2020-01-16 at 2 17 20 PM" src="https://user-images.githubusercontent.com/6620164/72555716-26b0d280-386b-11ea-945f-7401f6f68d71.png">
<img width="354" alt="Screen Shot 2020-01-16 at 2 17 42 PM" src="https://user-images.githubusercontent.com/6620164/72555717-26b0d280-386b-11ea-8a1c-10dadbad0b1c.png">
<img width="382" alt="Screen Shot 2020-01-16 at 2 20 11 PM" src="https://user-images.githubusercontent.com/6620164/72555803-54961700-386b-11ea-8fad-cb953bc105d4.png">

## Changes:
Updates Collection Page to handle Extra Memory Collections which:
- can not be renamed
- can not be deleted
- keep track of how many projects are in the collection
- disable the add project button when maxProjects has been reached
- give a warning when max projects has been reached
- give help text when there are no projects in the collection

A general note this does not update the Add Project popover to ensure that it only lets you add projects that are yours. That will require more backend work so I made separate tickets for it.

## How To Test:
* make a test account with a glitch.com email like name+test@glitch.com
* join the testing team: https://sarah-staging.glitch.me/@testing-team
* make a subscription by going to settings or by going to the pricing page
* click on the extra memory collection and make sure it works as outlined above ^
* make sure existing collection pages are not changed. 

## Feedback I'm looking for:
- any design thoughts?
- any feedback on tests? I'm trying to be better about it.
